health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/presentation/goals/ExerciseGoalsScreen.kt

```diff
@@ -0,0 +1,228 @@
+                        distanceValue = newValue
+                    }
+                }
+            }
+            if (!showDurationRow) {
+                item {
+                    Chip(onClick = { showDurationRow = !showDurationRow },
+                        label = { Text(stringResource(R.string.add_duration)) })
+                }
+            }
+            if (showDurationRow) {
+                item {
+                    DurationGoalRow(durationValue) { newValue -> durationValue = newValue }
+                }
+            }
+            item {
+                Row(verticalAlignment = Alignment.CenterVertically) {
+                    CompactChip(label = { Text(stringResource(R.string.set_goal)) },
+                        // Set the goal and pass the values to the view model
+                        onClick = {
+                            setGoals(
+                                distanceValue * 1000, (durationValue * 60).toLong()
+                            ); onSet() // Close the screen
+                        })
+                }
+            }
+        }
+    }
+
+
+}
+
+
+@Composable
+fun DistanceGoalRow(distanceValue: Double, onDistanceValueChange: (Double) -> Unit) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+        Text(
+            textAlign = TextAlign.Center,
+            text = stringResource(id = R.string.distance),
+            fontWeight = FontWeight.Bold
+        )
+        Spacer(modifier = Modifier.height(8.dp))
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            CompactButton(onClick = {
+                val newValue = distanceValue - 0.5
+                onDistanceValueChange(newValue)
+            }) {
+                Icon(
+                    imageVector = Icons.Default.Remove,
+                    contentDescription = stringResource(R.string.minus)
+                )
+            }
+            Text(
+                textAlign = TextAlign.Center,
+                text = distanceValue.toString() + " " + stringResource(R.string.km)
+            )
+            CompactButton(onClick = {
+                val newValue = distanceValue + 0.5
+                onDistanceValueChange(newValue)
+            }) {
+                Icon(
+                    imageVector = Icons.Default.Add,
+                    contentDescription = stringResource(R.string.plus)
+                )
+
+            }
+        }
+
+    }
+}
+
+@Composable
+fun DurationGoalRow(durationValue: Double, onDurationValueChange: (Double) -> Unit) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+        Text(
+            textAlign = TextAlign.Center,
+            text = stringResource(id = R.string.duration),
+            fontWeight = FontWeight.Bold
+        )
+
+        Spacer(modifier = Modifier.height(8.dp))
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            CompactButton(onClick = {
+                val newValue = durationValue - 5
```

Comment:
extract this out and call it something like durationGoalButtonDeltaMinutes

Reply:
Changed the goal setting format to a slider instead of buttons.